examples/demos/dnd.js

```diff
@@ -1,98 +0,0 @@
-import React from 'react';
-import BigCalendar from 'react-big-calendar';
-import { DragSource, DragDropContext, DropTarget } from 'react-dnd';
-import { findDOMNode } from 'react-dom';
-import HTML5Backend from 'react-dnd-html5-backend';
-import events from '../events';
-import merge from 'lodash/object/merge';
-
-/* drag sources */
-
-let eventSource = {
-  beginDrag: function (props) {
-    return props.event;
-  }
-}
-
-function collectSource(connect) {
-  return {
-    connectDragSource: connect.dragSource()
-  };
-}
-
-let DraggableEventWrapper = React.createClass({
-  render() {
-    let { connectDragSource, children, event } = this.props;
-    let EventWrapper = BigCalendar.components.eventWrapper;
-
-    return (<EventWrapper
-      event={event}
-      children={children}
-      ref={instance => connectDragSource(findDOMNode(instance))}
-    />);
-  }
-});
-
-DraggableEventWrapper = DragSource('event', eventSource, collectSource)(DraggableEventWrapper);
-
-/* drop targets */
-const dropTarget = {
-  drop(props, monitor, backgroundWrapper) {
-    const event = monitor.getItem();
-    alert(`${event.title} was dropped onto ${backgroundWrapper.props.type} ${backgroundWrapper.props.value}`);
-  }
-};
-
-function collectTarget(connect, monitor) {
-  return {
-    connectDropTarget: connect.dropTarget(),
-    isOver: monitor.isOver()
-  };
-}
-
-class DroppableBackgroundWrapper extends React.Component {
-  render() {
-    const { connectDropTarget, children, isOver } = this.props;
-    const BackgroundWrapper = BigCalendar.components.backgroundWrapper;
-
-    return (<BackgroundWrapper
-      children={children}
-      ref={instance => {
-        const domNode = findDOMNode(instance);
-        if(domNode) {
-          domNode.style.backgroundColor = isOver ? '#dddddd' : '';
-        }
-        return connectDropTarget(domNode);
-      }}
-    />);
-  }
-}
-
-DroppableBackgroundWrapper = DropTarget(['event'], dropTarget, collectTarget)(DroppableBackgroundWrapper);
-
-class DndBigCalendar extends React.Component {
-  propTypes: BigCalendar.propTypes
-
-  render() {
-    const propsCopy = merge({}, this.props);
-    propsCopy.components = merge({}, this.props.components, {
-      eventWrapper: DraggableEventWrapper,
-      backgroundWrapper: DroppableBackgroundWrapper
-    })
-    return <BigCalendar {...propsCopy} />
-  }
-}
-
-
-let Dnd = React.createClass({
```

Comment:
The example was originally split into the `DndBigCalendar` component and the `Dnd` export, to show that you could have a general `DndBigCalendar` that isn't tied to a specific set of events or dnd update behavior.  That way we could extract `DndBigCalendar` into a separate module (react-big-calendar addon) down the road.

Could you restore that structure in your code?  I would say that `moveEvent`, `defaultDate`, and `events` should be defined in the outer component, though you could perhaps have `DndBigCalendar` receive `moveEvent` as a prop (so `moveEvent` is implemented in `Dnd`, then passed to `DndBigCalendar` as a prop, and `DndBigCalendar` makes it available as part of its context).

With your nice separation of files, you could also have `DndBigCalendar` in a separate file - that would isolate the general component code from the code that uses it.